### ISSUE TITLE
[Bug] Kyuubi Spark Authz show databases/tables no data returned

### ISSUE THREAD

**Issue by lugela** [apache/incubator-kyuubi#5820]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

Kyuubi Spark Authz is connected to ranger1.2.0 and configured with relevant policies
![image](https://github.com/apache/kyuubi/assets/18652779/827af68b-3d79-4681-b0cb-95fe3b3bcff0)

This policy can be displayed normally on the spark-sql command line
![image](https://github.com/apache/kyuubi/assets/18652779/14f60828-d999-42f2-b9cf-0ca9315f39a6)

However, through access in kyuubi, the beeline/ldap method returns empty，the select specific strategy table can query data normally
<img width="890" alt="2ed9ba82b6dd225421e5359192c8a98" src="https://github.com/apache/kyuubi/assets/18652779/cd530dba-7fa5-4a8b-87e5-eebd7574c305">

### Affects Version(s)

1.8.0

### Kyuubi Server Log Output

```logtalk
no
```

### Kyuubi Engine Log Output

```logtalk
no
```

### Kyuubi Server Configurations

```yaml
no
```

### Kyuubi Engine Configurations

```yaml
no
```

### Additional context

no

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [X] No. I cannot submit a PR at this time.

**Comment by yaooqinn** [comment]
Did `beeline` and `spark-sql` have the same user identity?

**Comment by lugela** [comment]
> Did `beeline` and `spark-sql` have the same user identity?

yes ,It has been verified that it should be a bug in kyuubi 1.8.0. It is normal for kyuubi to fall back to 1.7.3

**Comment by eubnara** [comment]
I'm worried about the fact that getting ranger policy json files from ranger admin process to all the spark applications. It may be ddos attack to ranger admin process, isn't it?

**Comment by eubnara** [comment]
Also, I use both `cluster` and `client` for `spark.submit.deployMode` but `show databases` and `show tables` gives me empty result.
However, `select * from` query works.

**Comment by pan3793** [comment]
> I'm worried about the fact that getting ranger policy json files from ranger admin process to all the spark applications. It may be ddos attack to ranger admin process, isn't it?

Definitely, especially when you have a lot of policies, TBH, it's a limitation of Ranger's architecture, AFAIK, there are many users extend the Ranger to support the server-side policy evaluation to address such issues.

**Comment by eubnara** [comment]
Thanks for your opinion, I think if authorization only occurs on Kyuubi server side and uses spark just as engine like HiveServer2, the problem will be alleviated.
